File: server/src/seed.js
import 'dotenv/config';
import bcrypt from 'bcryptjs';
import { connectDB } from './db.js';
import User from './models/User.js';
import Vehicle from './models/Vehicle.js';
import Trip from './models/Trip.js';
import Maintenance from './models/Maintenance.js';

export async function seedInitialData(force = false) {
  if (!force) {
    const count = await User.countDocuments();
    if (count > 0) return;
  }
  await Promise.all([User.deleteMany({}), Vehicle.deleteMany({}), Trip.deleteMany({}), Maintenance.deleteMany({})]);
  const admin = await User.create({ name: 'Fleet Admin', email: 'admin@fleet.com', password: await bcrypt.hash('Admin@123',10), role: 'admin', phone: '9000000001' });
  const driver = await User.create({ name: 'Ravi Kumar', email: 'driver@fleet.com', password: await bcrypt.hash('Driver@123',10), role: 'driver', phone: '9000000002' });
  const d2 = await User.create({ name: 'Priya Sharma', email: 'priya@fleet.com', password: await bcrypt.hash('Driver@123',10), role: 'driver', phone: '9000000003' });
  const today = new Date();
  const daysAgo = n => new Date(Date.now() - n*86400000);
  const v1 = await Vehicle.create({ registrationNumber:'AP39AB1234', type:'Delivery Van', purchaseDate:daysAgo(900), lastServiceDate:daysAgo(25), currentOdometer:45200, status:'available', assignedDriver:null });
  const v2 = await Vehicle.create({ registrationNumber:'AP40CD5678', type:'Field Service Car', purchaseDate:daysAgo(600), lastServiceDate:daysAgo(105), currentOdometer:38100, status:'available', assignedDriver:null });
  const v3 = await Vehicle.create({ registrationNumber:'AP41EF9012', type:'Pickup Truck', purchaseDate:daysAgo(1200), lastServiceDate:daysAgo(45), currentOdometer:67000, status:'available' });
  await Trip.create([
   { driver:driver._id, vehicle:v1._id, startLocation:'Vijayawada', endLocation:'Guntur', startTime:daysAgo(2), endTime:new Date(Date.now()-2*86400000+2*3600000), distanceKm:65, fuelUsedLitres:8 },
   { driver:driver._id, vehicle:v1._id, startLocation:'Guntur', endLocation:'Tenali', startTime:daysAgo(1), endTime:new Date(Date.now()-86400000+90*60000), distanceKm:48, fuelUsedLitres:6 },
   { driver:d2._id, vehicle:v2._id, startLocation:'Visakhapatnam', endLocation:'Anakapalle', startTime:daysAgo(3), endTime:new Date(Date.now()-3*86400000+2*3600000), distanceKm:70, fuelUsedLitres:7.5 }
  ]);
  await Maintenance.create([
   { vehicle:v1._id, serviceDate:daysAgo(25), serviceType:'Oil & Filter', cost:3200, nextServiceDate:new Date(Date.now()+65*86400000), notes:'Routine service' },
   { vehicle:v2._id, serviceDate:daysAgo(105), serviceType:'Full Service', cost:8500, nextServiceDate:today, notes:'Service overdue' },
   { vehicle:v3._id, serviceDate:daysAgo(45), serviceType:'Brake Inspection', cost:1800, nextServiceDate:new Date(Date.now()+45*86400000) }
  ]);
  console.log('Seed complete');
}

if (process.argv[1] && process.argv[1].endsWith('seed.js')) {
  await connectDB();
  await seedInitialData(true);
  process.exit(0);
}

File: flask_server/seed.py
from datetime import datetime, timedelta, timezone
import bcrypt
from models import db, User, Vehicle, Trip, Maintenance

def seed_database():
    db.drop_all()
    db.create_all()

    pw_admin = bcrypt.hashpw('Admin@123'.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
    pw_driver = bcrypt.hashpw('Driver@123'.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')

    admin = User(name='Fleet Admin', email='admin@fleet.com', password=pw_admin, role='admin', phone='9000000001')
    driver = User(name='Ravi Kumar', email='driver@fleet.com', password=pw_driver, role='driver', phone='9000000002')
    driver2 = User(name='Priya Sharma', email='priya@fleet.com', password=pw_driver, role='driver', phone='9000000003')

    db.session.add_all([admin, driver, driver2])
    db.session.commit()

    now = datetime.now(timezone.utc).replace(tzinfo=None)

    v1 = Vehicle(
        registration_number='AP39AB1234',
        type='Delivery Van',
        purchase_date=now - timedelta(days=900),
        last_service_date=now - timedelta(days=25),
        current_odometer=45200,
        status='available',
        assigned_driver_id=None,
        current_load='Empty / Unloaded',
        load_weight_kg=0.0,
        load_status='unloaded'
    )
    v2 = Vehicle(
        registration_number='AP40CD5678',
        type='Field Service Car',
        purchase_date=now - timedelta(days=600),
        last_service_date=now - timedelta(days=105),
        current_odometer=38100,
        status='available',
        assigned_driver_id=None,
        current_load='Empty / Unloaded',
        load_weight_kg=0.0,
        load_status='unloaded'
    )
    v3 = Vehicle(
        registration_number='AP41EF9012',
        type='Pickup Truck',
        purchase_date=now - timedelta(days=1200),
        last_service_date=now - timedelta(days=45),
        current_odometer=67000,
        status='available',
        current_load='Empty / Unloaded',
        load_weight_kg=0.0,
        load_status='unloaded'
    )

    db.session.add_all([v1, v2, v3])
    db.session.commit()

    t1 = Trip(
        driver_id=driver.id,
        vehicle_id=v1.id,
        start_location='Vijayawada',
        end_location='Guntur',
        start_time=now - timedelta(days=2),
        end_time=now - timedelta(days=2) + timedelta(hours=2),
        distance_km=65,
        fuel_used_litres=8,
        cargo_details='Electronics Consignment (450 kg)',
        returned_vehicle=False
    )
    t2 = Trip(
        driver_id=driver.id,
        vehicle_id=v1.id,
        start_location='Guntur',
        end_location='Tenali',
        start_time=now - timedelta(days=1),
        end_time=now - timedelta(days=1) + timedelta(minutes=90),
        distance_km=48,
        fuel_used_litres=6,
        cargo_details='Retail Electronics Parcel',
        returned_vehicle=False
    )
    t3 = Trip(
        driver_id=driver2.id,
        vehicle_id=v2.id,
        start_location='Visakhapatnam',
        end_location='Anakapalle',
        start_time=now - timedelta(days=3),
        end_time=now - timedelta(days=3) + timedelta(hours=2),
        distance_km=70,
        fuel_used_litres=7.5,
        cargo_details='Hardware Supplies (120 kg)',
        returned_vehicle=False
    )

    m1 = Maintenance(
        vehicle_id=v1.id,
        service_date=now - timedelta(days=25),
        service_type='Oil & Filter',
        cost=3200,
        next_service_date=now + timedelta(days=65),
        notes='Routine service'
    )
    m2 = Maintenance(
        vehicle_id=v2.id,
        service_date=now - timedelta(days=105),
        service_type='Full Service',
        cost=8500,
        next_service_date=now,
        notes='Service overdue'
    )
    m3 = Maintenance(
        vehicle_id=v3.id,
        service_date=now - timedelta(days=45),
        service_type='Brake Inspection',
        cost=1800,
        next_service_date=now + timedelta(days=45)
    )

    db.session.add_all([t1, t2, t3, m1, m2, m3])
    db.session.commit()
    print("Python MySQL Database Seed Complete!")

if __name__ == '__main__':
    from app import create_app
    app = create_app()
    with app.app_context():
        seed_database()
